lumigator/python/mzai/backend/backend/services/jobs.py

```diff
@@ -130,12 +132,13 @@ def _add_dataset_to_db(self, job_id: UUID, request: JobInferenceCreate, s3: S3Fi
         # Get the dataset from the S3 bucket
         result_key = self._get_results_s3_key(job_id)
         with s3.open(f"{settings.S3_BUCKET}/{result_key}", "r") as f:
-            results = json.loads(f.read())
+            results = InferenceJobOutput.model_validate(json.loads(f.read()))
+        # Validate that the output file adheres to
```

Comment:
nit: let's leave the commend above the `results`, would also be helpful for legibility to type results